src/dispensary/updater.js

```diff
@@ -0,0 +1,249 @@
+/* eslint  import/no-extraneous-dependencies: 0 */
```

Comment:
Instead of disabling the `import/no-extraneuous-dependencies` eslint rule on this file, I would slightly prefer it to live outside of the `src/` directory.

From a bundling perspective it does not make a real difference, webpack treeshaking behavior would be omitting this module entirely from the bundle because none of the src modules being bundled does import this one.

Nevertheless, moving this out of `src/` would make it crystal clear that it is not something to import from another addons-linter source.

We could either:
- move it in the form of an ESM (almost as is) into `scripts/lib` and make the script to import it from there.

- Or we could merge it with `scripts/dispensary` itself if we prefer

Personally I'm fine either way.

------

Merging it with `scripts/dispensary` seems also doable, e.g. by making it to not behave like cli script when imported as a module from the unit tests:
```
if (require.main === module) {
  new Updater().run().catch((err) => {
    console.error('Uncaught error', err);
  });
}
```

but most of the changes to merge it into the `script/dispensary` script may likely be required due to the need to adapt it to a commonjs module, which may be more or less:
```
const path = require('path');
const fs = require('fs');

const async = require('async');
const fetch = require('node-fetch');
const naturalCompare = require('natural-compare-lite');

const { default: log } = require('../src/logger');
const { default: createHash } = require('../src/dispensary/hasher');
const { urlFormat } = require('../src/dispensary/utils');

class Updater {
  ...
}

module.exports = { Updater };
...
```

In both cases the path to the files would need to change to be relative to the `scripts/lib` or `scripts` directory (`DEFAULT_PATH_TO_HASHES` and `DEFAULT_LIBRARY_FILE`).

and I guess also one change on the unit test side:
```
import { Updater } from '../../../scripts/dispensary';
```

Everything I think that would still work by staying exactly the same as in this version.

Reply:
Yeah, I think I am good with that. I suppose we can merge the two files.